### DISCUSSION TITLE
PIO core version selection

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#1550] [General]
in chapter 5.4 of the core's docu it is stated that one should not anymore use 

```
; note that download link for toolchain is specific for OS. see https://github.com/
˓→earlephilhower/pico-quick-toolchain/releases.
platform_packages =
maxgerhardt/framework-arduinopico@https://github.com/earlephilhower/arduino-pico.git
maxgerhardt/toolchain-pico@https://github.com/earlephilhower/pico-quick-toolchain/
˓→releases/download/1.3.1-a/x86_64-w64-mingw32.arm-none-eabi-7855b0c.210706.zip
```

we use currently this lines to stay on core 2.7.3 to prevent any changes in the core influence our fw.
https://github.com/OpenKNX/OGM-Common/blob/modular/platformio.base.ini#L50C9-L53

while it works for now and we are thinking now to switch to core 3.3.0 the question arises what to do to keep core 3.3.0 in a reproducable and not deprecated way..

@maxgerhardt you have some hint for us?

**Comment by maxgerhardt** [comment]
That way of doing it is still okay for purposes of reproducability. Before the packages were in the registry, that was **needed**, but not now it's not *needed* anymore for regular "just use the latest version" purposes, but still perfectly fine for fixing versions.

Just note that if you want to fix the toolchain, the package is now called `toolchain-rp2040-earlephilhower`, not `toolchain-pico`. It can be sourced from the same https://github.com/earlephilhower/pico-quick-toolchain/ toolchain though. Same goes for `tool-mklittlefs-rp2040-earlephilhower` if your project explicitly uses the LittleFS `mklittlefs` tool / `data` folder upload.

**Comment by Ing-Dom** [comment]
Thanks for the quick reply! To be honest I do not fully understand what all these options exactly do.

I thought all neccessary tools ship with the core?
Do I need something in addition to that?
```
platform = https://github.com/maxgerhardt/platform-raspberrypi.git#c0acac7
platform_packages = framework-arduinopico @ https://github.com/earlephilhower/arduino-pico/releases/download/3.3.0/rp2040-3.3.0.zip
```
like 
toolchain-rp2040-earlephilhower @ https://github.com/earlephilhower/pico-quick-toolchain.git#1.5.0-b
